cloud/rackspace/rax_clb_ssl.py

```diff
@@ -0,0 +1,259 @@
+        else:
+            certificate = certificate.strip()
+
+    attempts = wait_timeout / 5
+
+    # Locate the load balancer.
+
+    balancer = rax_find_loadbalancer(module, pyrax, loadbalancer)
+    existing_ssl = balancer.get_ssl_termination()
+
+    changed = False
+
+    if state == 'present':
+        # Apply or reconfigure SSL termination on the load balancer.
+        ssl_attrs = dict(
+            securePort=secure_port,
+            privatekey=private_key,
+            certificate=certificate,
+            intermediateCertificate=intermediate_certificate,
+            enabled=enabled,
+            secureTrafficOnly=secure_traffic_only
+        )
+
+        needs_change = False
+
+        if existing_ssl:
+            for ssl_attr, value in ssl_attrs.iteritems():
+                if ssl_attr == 'privatekey':
+                    # The private key is not included in get_ssl_termination's
+                    # output (as it shouldn't be). Also, if you're changing the
+                    # private key, you'll also be changing the certificate,
+                    # so we don't lose anything by not checking it.
+                    continue
+
+                if value is not None and existing_ssl.get(ssl_attr) != value:
+                    # module.fail_json(msg='Unnecessary change', attr=ssl_attr, value=value, existing=existing_ssl.get(ssl_attr))
+                    needs_change = True
+        else:
+            needs_change = True
+
+        if needs_change:
+            balancer.add_ssl_termination(**ssl_attrs)
+            changed = True
+    elif state == 'absent':
+        # Remove SSL termination if it's already configured.
+        if existing_ssl:
+            balancer.delete_ssl_termination()
+            changed = True
+
+    if https_redirect is not None and balancer.httpsRedirect != https_redirect:
+        if changed:
+            # This wait is unavoidable because load balancers are immutable
+            # while the SSL termination changes above are being applied.
+            pyrax.utils.wait_for_build(balancer, interval=5, attempts=attempts)
+
+        balancer.update(httpsRedirect=https_redirect)
+        changed = True
+
+    if changed and wait:
+        pyrax.utils.wait_for_build(balancer, interval=5, attempts=attempts)
+
+    balancer.get()
+    new_ssl_termination = balancer.get_ssl_termination()
+
+    # Intentionally omit the private key from the module output, so you don't
+    # accidentally echo it with `ansible-playbook -v` or `debug`, and the
+    # certificate, which is just long. Convert other attributes to snake_case
+    # and include https_redirect at the top-level.
+    if new_ssl_termination:
+        new_ssl = dict(
+            enabled=new_ssl_termination['enabled'],
+            secure_port=new_ssl_termination['securePort'],
+            secure_traffic_only=new_ssl_termination['secureTrafficOnly']
+        )
+    else:
+        new_ssl = None
+
+    result = dict(
```

Comment:
I think the `result` should also include some data about what load balancer it just acted on.